auth/src/main/kotlin/com/hedvig/android/auth/AuthTokenService.kt

```diff
@@ -6,9 +6,9 @@ import kotlinx.coroutines.flow.StateFlow
 
 interface AuthTokenService {
   val authStatus: StateFlow<AuthStatus?>
-  fun getToken(): AccessToken?
-  suspend fun refreshAndGetToken(): AccessToken?
-  fun updateTokens(accessToken: AccessToken, refreshToken: RefreshToken)
-  fun invalidateTokens()
+  suspend fun getTokens(): AuthTokens?
+  suspend fun refreshAndGetAccessToken(): AccessToken?
+  suspend fun updateTokens(accessToken: AccessToken, refreshToken: RefreshToken)
```

Comment:
Let this take AuthTokens as paremeter as well?

Reply:
I liked the idea, but I went ahead and tried to do it and realized that `AuthTokens` takes in the `Local` versions of the tokens, which means that ones with the expiration as an `Instant` instead of an `int` with seconds to expire from now, so this wouldn't work. Since we don't want to do this conversion in all call sites, and keep that logic inside `AuthTokenStorage` so gonna keep it like this, and let this take in the types that authlib returns directly, to not overload this with another `AuthTokens` version which is for those versions of the tokens.